addons/mail/models/mail_thread.py

```diff
@@ -1142,7 +1145,18 @@ def _message_route_process(self, message, message_dict, routes):
             else:
                 # if a new thread is created, parent is irrelevant
                 message_dict.pop('parent_id', None)
-                thread = ModelCtx.message_new(message_dict, custom_values)
+                # Report failure/record success of message creation except if alias is not defined (fallback model case)
+                try:
+                    thread = ModelCtx.message_new(message_dict, custom_values)
+                except Exception:
+                    if alias:
+                        with self.pool.cursor() as new_cr:
+                            self.with_env(self.env(cr=new_cr)).env['mail.alias'].browse(alias.id) \
+                                ._set_alias_invalid(message, message_dict)
+                    raise
+                else:
+                    if alias and alias.alias_status != 'valid':
+                        alias.sudo().alias_status = 'valid'
```

Comment:
If we removed the sudo in set_alias_invalid, we probably don't need it here, do we ?

Reply:
Yes indeed; tested, it works but it causes a problem in the mass mailing test which simulate response to a mailing configured with replies to an alias. The problem is that message_process in those tests are not called with admin rights and it fails when trying to set the alias status to valid. I have sudoed that operation in common.py. What do you think ?